### DISCUSSION TITLE
Check if the current request is a request via DataHub GraphQL API

### DISCUSSION THREAD

**Discussion by Blackbam** [pimcore/pimcore#9253] [General]
For me it is not trivial to find a way if the current request is

a) a request via Pimcore DataHub API
b) if the request passed the configuration check and permissions check

What I want to do is to create a Service function as simple as:

```
function isValidDataHubRequest() {}
```

What is the correct way to achieve this?

**Comment by weisswurstkanone** [comment]
a) maybe attach to the ExecutorEvents::PRE_EXECUTE event at set it to the runtime. See https://github.com/pimcore/data-hub/blob/ec5e1bd1ee89e8461a223dc69c05596fca28164b/src/Controller/WebserviceController.php#L183
b) permissions are checked anyway
